server/src/main/java/org/elasticsearch/action/admin/indices/alias/TransportIndicesAliasesAction.java

```diff
@@ -161,18 +165,21 @@ protected void masterOperation(
                                 finalActions.add(new AddDataStreamAlias(alias, dataStreamName, action.writeIndex(), action.filter()));
                             }
                         }
+                        actionResults.add(AliasActionResult.build(action, numAliasesRemoved));
```

Comment:
This doesn't hurt anything, but it threw me off for a second that we're passing `numAliasesRemoved` when building an `add` action result. AliasActionResult.build correctly ignores it.

Reply:
Yeah, that does look weird. I think it's worth using separate `build` and `buildRemoveAlias` builder functions, so it's a bit clearer.